csrc/device_lower/pass/index.cpp

```diff
@@ -1816,6 +1821,230 @@ Val* hardCodedIndexGenerationForStMatrix(
   return out;
 }
 
+// Goal: Store (tma_m, tma_n) row-major tile in shared memory using stmatrix
+// (stsm_m_tile, stsm_n_tile)
+//
+// Let shared_memory tile be (tma_m = 64, tma_n = 128).
+// Let StMatrix tile be (stsm_m_tile = 16, stsm_n_tile = 16).
+// Let dtype be fp16 or bf16, so dtype_size is 2B.
+//
+// To avoid shared memory bank conflicts, apply swizzle on stmatrix and
+// tma store operations. Let swizzle be MmaInputSmemSwizzle::B128.
```

Comment:
> Let swizzle be MmaInputSmemSwizzle::B128.

That means in this example we assume the swizzle is 128 bytes right? Not that we are only supporting 128B.

Reply:
This function only supports 128B swizzle, but the transformation is necessarily limited to that swizzle.